satpy/composites/__init__.py

```diff
@@ -1100,9 +1131,27 @@ def _get_and_sharpen_rgb_data_arrays_and_meta(self, datasets, optional_datasets)
                 chunks=high_res.chunks,
             )
             with xr.set_options(keep_attrs=True):
-                low_res_red = high_res if low_resolution_index == 0 else low_res_red * ratio
-                low_res_green = high_res if low_resolution_index == 1 else low_res_green * ratio
-                low_res_blue = high_res if low_resolution_index == 2 else low_res_blue * ratio
+                if neutral_res is not None:
+                    if high_resolution_index == 0:
+                        low_res_red = high_res
+                        low_res_green = neutral_res if neutral_resolution_index == 1 else low_res_green * ratio
+                        low_res_blue = neutral_res if neutral_resolution_index == 2 else low_res_blue * ratio
+
+                    elif high_resolution_index == 1:
+                        low_res_red = neutral_res if neutral_resolution_index == 0 else low_res_red * ratio
+                        low_res_green = high_res
+                        low_res_blue = neutral_res if neutral_resolution_index == 2 else low_res_blue * ratio
+
+                    elif high_resolution_index == 2:
+                        low_res_red = neutral_res if neutral_resolution_index == 0 else low_res_red * ratio
+                        low_res_green = neutral_res if neutral_resolution_index == 1 else low_res_green * ratio
+                        low_res_blue = high_res
+
+                else:
+                    low_res_red = high_res if high_resolution_index == 0 else low_res_red * ratio
+                    low_res_green = high_res if high_resolution_index == 1 else low_res_green * ratio
+                    low_res_blue = high_res if high_resolution_index == 2 else low_res_blue * ratio
```

Comment:
What if we put this ternary operation (`X if index == Y else Z * ratio`) in a separate method and just called it for each band:

```
low_res_red = self._choose_sharpest_band(low_res_red, high_res, high_resolution_index, ratio)
```

Ugh nevermind, that doesn't handle the neutral band check...but that's ugly too. Oh but it also seems like you added the `high_resolution_index` and stopped using `low_resolution_index`. That shouldn't be needed, right?

I'm really unsure of this block of if/else statements. There has to be a cleaner way, but I'm having trouble thinking of it.

Reply:
I think the main confusion here is that `neutral_res` doesn't need to be treated as an "action" that we are performing on the band. We are just not doing anything extra to that band. In the case of the high res band/index, we are swapping. I'm still not exactly sure how to clean the logic up though.